Client/Frontend/Browser/SearchViewController.swift

```diff
@@ -325,6 +325,13 @@ class SearchViewController: SiteTableViewController, KeyboardHelperDelegate, Loa
         animateSearchEnginesWithKeyboard(state)
     }
 
+    override func viewWillTransitionToSize(size: CGSize, withTransitionCoordinator coordinator: UIViewControllerTransitionCoordinator) {
+        // The height of the suggestions row may change, so call reloadData() to recalculate cell heights.
+        coordinator.animateAlongsideTransition({ _ in
+            self.tableView.reloadData()
```

Comment:
It will be called on the UI thread (all UI-related methods/callbacks must happen on the UI thread). From testing, the reload happens right before the animation starts. Another option would be to move it to the completion block after the animation ends, though the reload is more noticeable and looks jankier.

I don't think `systemLayoutFittingSizeDidChangeForChildContentContainer` would be an improvement since `viewWillTransitionToSize` already works about as well as we could hope.